src/controller/AutoCommissioner.cpp

```diff
@@ -197,13 +197,28 @@ CHIP_ERROR AutoCommissioner::SetCommissioningParameters(const CommissioningParam
             if (mTimeZoneBuf[i].name.HasValue())
             {
                 auto span = MutableCharSpan(mTimeZoneNames[i], kMaxTimeZoneNameLen);
+                // This buffer is statically allocated and if of size kMaxSupportedTimeZones, so this should never fail
                 CopyCharSpanToMutableCharSpan(mTimeZoneBuf[i].name.Value(), span);
                 mTimeZoneBuf[i].name.SetValue(span);
             }
         }
         auto list = app::DataModel::List<app::Clusters::TimeSynchronization::Structs::TimeZoneStruct::Type>(mTimeZoneBuf, size);
         mParams.SetTimeZone(list);
     }
+    if (params.GetDefaultNTP().HasValue()) {
+        ChipLogProgress(Controller, "Setting Default NTP from parameters");
+        // This parameter is an optional nullable, so we need to go two levels deep here.
+        if (!params.GetDefaultNTP().Value().IsNull()) {
+            size_t size = std::min(params.GetDefaultNTP().Value().Value().size(), kMaxDefaultNtpSize);
+            // This buffer is statically allocated and is of size kMaxDefaultNtpSize.
+            auto span = MutableCharSpan(mDefaultNtp, kMaxDefaultNtpSize);
+            CopyCharSpanToMutableCharSpan(params.GetDefaultNTP().Value().Value(), span);
```

Comment:
This copy will fail if `params.GetDefaultNTP().Value().Value().size() > kMaxDefaultNtpSize`, no?   And then we have garbage in `span`....

What we might want to do is grab `params.GetDefaultNTP().Value().Value()` into an on-stack Span, figure out the size we want to copy, take a subspan of that size, and copy that....

Reply:
As above - I think we need a different approach here because a partial string is basically useless. Going to do something different, and I THINK I can add a unit test.